src/store/user_recent_tags.js

```diff
@@ -0,0 +1,37 @@
+/*
+ This file is a part of libertysoil.org website
+ Copyright (C) 2016  Loki Education (Social Enterprise)
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import i from 'immutable';
+
+import * as a from '../actions';
+
+const initialState = i.fromJS({
+  geotags: [],
+  schools: [],
+  hashtags: []
+});
+
+export default function reducer(state=initialState, action) {
+  switch (action.type) {
+    case a.SET_USER_RECENT_TAGS: {
+      state = state.merge(action.user_recent_tags);
+      break;
+    }
+  }
+
+  return state;
+}
```

Comment:
Move this to `store/current-user` please. And don't forget to handle logout (clean "recent" on logout)